### DISCUSSION TITLE
How to access contents of a webview?

### DISCUSSION THREAD

**Discussion by iSuslov** [tauri-apps/tauri#9626] [Q&A]
I don't see an example of accessing Webview contents. Via injecting script or directly, is it something Tauri supports?

Example. Let's say I've created a webview using JSX syntax:

```tsx
declare global {
    interface Window { webview: Webview; }
}

function App() {
  useEffect(() => {
    if (window.webview) {
      return;
    }
    const currWindow = getCurrent();
    const webview = new Webview(currWindow, "theUniqueLabel", {
      x: 10,
      y: 10,
      width: 400,
      height: 400,
      url: "https://github.com/tauri-apps/tauri",
    });
    webview.once("tauri://created", function (e) {
      // webview successfully created
      console.log("CREATED", e);
    });
    webview.once("tauri://error", function (e) {
      // an error happened creating the webview
      console.log("ERROR", e);
    });
    window.webview = webview;
  }, []);
  return <div></div>
}
```  

It seems like I don't have much of a c control over `Webview` contents. How can I communicate with it? Inject script? Anything similar to Electron `<webview>` tag / Webview class functionality: https://www.electronjs.org/docs/latest/api/webview-tag

**Comment by edbrdi** [comment]
You can create and inject JavaScript into a WebView on the Rust side, and invoke this code from the client side. However, I was unable to establish communication from the WebView back to the client. The solution I found for now is to run a local HTTP server (using actix_web) when the app is launched, and then have the JavaScript sent to the WebView create a form to submit data back to the server. It works, but it's not very convenient...
